app/pages/admin/project-status.jsx

```diff
@@ -76,10 +79,28 @@ class ProjectStatus extends Component {
   handleToggle(event, workflow) {
     this.setState({ error: null });
     const checked = event.target.checked;
+    const defaultWorkflowId = this.state.project.configuration.default_workflow;
+
+    const promises = [this.getWorkflows(), this.getProject()];
+
+    if (defaultWorkflowId === workflow.id && workflow.active) {
+      Dialog.alert(
+        <WorkflowDefaultDialog closeButton={true} required={true} />
+      )
+      .then(() => {
+        Promise.all(promises)
+          .then(() => {
+            if (!this.state.error) {
```

Comment:
If there is an error, then the promise is rejected, which is what `catch()` is for.